myokit/_model_api.py

```diff
@@ -2933,7 +2933,9 @@ def set_state(self, state):
         Changes this model's state. Accepts any type of input handled by
         :meth:`map_to_state`.
         """
-        self._current_state = self.map_to_state(state)
+        self._current_state = [
+            myokit.Number(x) for x in self.map_to_state(state)
```

Comment:
It **might** make sense to update `map_to_state` to returns Number objects.

Reply:
Quick grep:
```
./myokit/_sim/cvodesim.py:            y = self._model.map_to_state(y)
./myokit/_sim/cvodesim.py:        self._default_state = self._model.map_to_state(state)
./myokit/_sim/cvodesim.py:        self._state = self._model.map_to_state(state)
./myokit/_sim/cvodessim.py:            y = self._model.map_to_state(y)
./myokit/_sim/cvodessim.py:        self._default_state = self._model.map_to_state(state)
./myokit/_sim/cvodessim.py:        self._state = self._model.map_to_state(state)
./myokit/_io.py:            s = model.map_to_state(s)
./myokit/_io.py:        state = model.map_to_state(state)
./myokit/_model_api.py:            self.set_state(self.map_to_state(state))
./myokit/_model_api.py:        self._current_state = self.map_to_state(state)
./myokit/__main__.py:        ref = model.map_to_state(ref)
```

So maybe not

Reply:
agree maybe leave it for now?

Reply:
yeah